docs/development/privacy.md

```diff
@@ -0,0 +1,49 @@
+# Privacy Checklist for ILIAS Development
```

Comment:
Please consider stating the target group of the check list. Is it developers or admins? Please consider asking open ended questions to reflect the purpose-tied and risk-based approach of the GDPR.